tauri-app/src-tauri/src/wine_manager.rs

```diff
@@ -53,26 +53,55 @@ impl WineManager {
         .spawn().expect("Failed to init prefix").wait().expect("Failed to init prefix");
     }
 
-    pub fn launch_exe(&self, absolute_path: &str, args: &Vec<&str>) -> std::io::Result<()> {
+    pub fn launch_exe(&self, absolute_path: &str, vars: &Vec<(&str, &str)>, args: &Vec<&str>) -> std::io::Result<()> {
         let display = env::var("DISPLAY").unwrap_or_else(|_| ":0".to_string());
         let mut command = Command::new("wine");
-        command.env("WINEPREFIX", &self.prefix).env("DISPLAY", display);
         command.arg(absolute_path);
+
+        for (var, value) in vars {
+            command.env(var, value);
+        }
         for arg in args {
             command.arg(arg);
         }
 
-        command.spawn()?.wait()?;
+        command
+            .env("WINEPREFIX", &self.prefix)
+            .env("DISPLAY", display)
+            .spawn()?
+            .wait()?;
+
         Ok(info!("Launched {absolute_path}"))
     }
 
-    pub(crate) fn load_cd(&self, source_dir: &str, drive_letter: &str) {
-        let source_path = self.get_c_path(source_dir);
+    pub(crate) fn load_cd(&self, source_dir: &str, drive_letter: &str) -> io::Result<()> {
+        let source_dir = format!("../drive_c/{}", source_dir);
         let drive_path = self.get_drive_path(drive_letter);
-        os::unix::fs::symlink(source_path, drive_path).unwrap();
+        if drive_path.exists() {
+            if !drive_path.is_symlink() {
+                return Err(io::Error::new(io::ErrorKind::AlreadyExists,
+                    format!("{:?} already exists and is not a symlink!", drive_path.to_str())));
+            }
+
+            let existing_link = std::fs::read_link(drive_path)?;
+            let existing_link = existing_link.to_str()
+            .ok_or(io::Error::new(io::ErrorKind::InvalidData, "Failure to read existing link!"))?;
+
+            if existing_link != source_dir {
```

Comment:
I think this will fail, if someone changes their username. I think we can clean this up later though, as alot of the error handling needs a proper think

Reply:
I don't think it will because the symlink is relative. You can make a symlink on your own system that points to a relative path and if you move that symlink, it will break the link rather than adjust the path it is pointing to.

Reply:
I reworked the error handling here to just remove any symlink or file that exists there if it's not a symlink to the folder we are trying to mount. This should be fine as long as we don't start mounting other folders as drives and accidentally use the same one twice.